openmetadata-ui/src/main/resources/ui/src/components/buttons/CopyToClipboardButton/CopyToClipboardButton.tsx

```diff
@@ -33,44 +31,32 @@ export const CopyToClipboardButton: FunctionComponent<Props> = ({
   onCopy,
 }: Props) => {
   const { t } = useTranslation();
-  const [copied, setCopied] = useState<boolean>(false);
-
-  const handleCopying = () => {
-    setCopied(true);
-    onCopy?.();
-    setTimeout(() => {
-      setCopied(false);
-    }, copyTimer);
-  };
+  const { hasCopied, onCopyToClipBoard } = useClipboard(
```

Comment:
Can we improve unit tests for this component?
Interaction and negative cases?

1. Callback should have been called
2. After timeout tooltip should be hidden
3. Clipboard must have text once hit copy
4. What happen if we don't have access to navigator APIs?

Reply:
Sure @chirag-madlani , I will work on it.

Reply:
@chirag-madlani , I have pushed the changes, please review.